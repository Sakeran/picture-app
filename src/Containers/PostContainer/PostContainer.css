.PostContainer-display {
  text-align: center;
  margin-bottom: 1em;
}

.PostContainer-img {
  max-width: 100%;
  height: auto;
}

.PostContainer-details {
  margin-left: 1em;
}

.PostContainer-details p {
  font-size: 0.9em;
  margin-left: 1em;
}

.PostContainer-comments {
  text-align: center;
}
