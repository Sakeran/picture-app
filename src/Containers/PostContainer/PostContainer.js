import React from 'react';
import PropTypes from 'prop-types';

import SafeImage from '../../Components/SafeImage/SafeImage';
import PostDetails from '../../Components/PostDetails/PostDetails';
import PostStats from '../../Components/PostStats/PostStats';

import { graphql, gql, compose} from 'react-apollo';

import './PostContainer.css';

class PostContainer extends React.Component {

  render() {
    const { data: {loading, error, post}} = this.props;
    if (loading) {
      return <h2>Loading...</h2>;
    }
    if (error) {
      return <h2>Error While Loading Post</h2>;
    }
    const postDetails = {
      title: post.title,
      description: post.description,
      postDate: post.postDate,
      creator: post.creator
    };
    const postStats = {
      likeCount: post.likeCount,
    };
    return (
      <div className="PostContainer">
        <div className="PostContainer-display">
          <SafeImage className="PostContainer-img" src={post.image} alt={post.title} />
        </div>
        <PostDetails {...postDetails} />
        <PostStats {...postStats}/>
        <div className="PostContainer-comments">
          <p>Comments go here</p>
        </div>
      </div>
    );
  }

};

PostContainer.propTypes = {
  postId: PropTypes.string.isRequired
};

const postDetailsQuery = gql`
  query postDetails($id: ID!) {
    post(id: $id) {
      id
      type
      title
      description
      postDate
      image
      youtubeID
      creator {
        id
        username
      }
      likeCount
      commentCount
    }
  }
`;

export { PostContainer };
export default compose(
  graphql(postDetailsQuery, {
    options: (props) => ({
      variables: {
        id: props.postId
      }
    })
  })
)(PostContainer);
