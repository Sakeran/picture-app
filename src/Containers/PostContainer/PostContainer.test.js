import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PostContainer } from './PostContainer';

const makePost = (id, type) => ({
  id: id,
  type: type || 'image',
  image: 'http://localhost:3001/exampleImg.png',
  title: 'test card ' + id,
  youtubeID: 'xxxxxxxxxxx',
  creator: {
    id: '100',
    username: 'testUser'
  }
});

test('it renders correctly', () => {
  const data = {
    post: makePost('1')
  }
  const component = shallow(<PostContainer postId="1" data={data} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
