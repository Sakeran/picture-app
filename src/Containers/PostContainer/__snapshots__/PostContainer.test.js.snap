// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="PostContainer"
>
  <div
    className="PostContainer-display"
  >
    <SafeImage
      alt="test card 1"
      className="PostContainer-img"
      src="http://localhost:3001/exampleImg.png"
    />
  </div>
  <PostDetails
    creator={
      Object {
        "id": "100",
        "username": "testUser",
      }
    }
    description="Test post."
    postDate="May 1st 2000"
    title="test card 1"
  />
  <PostStats />
  <div
    className="PostContainer-comments"
  >
    <p>
      Comments go here
    </p>
  </div>
</div>
`;
